overlord/snapstate/refreshhints.go

```diff
@@ -261,6 +261,36 @@ func pruneRefreshCandidates(st *state.State, snaps ...string) error {
 	return nil
 }
 
+// updateRefreshCandidates updates the current set of refresh candidates stored
+// in the state in such a way that entries which are not present currently get
+// added, but ones that do have their snap-setup updated, so that it reflects
+// the latest information obtained from the store
+func updateRefreshCandidates(st *state.State, hints map[string]*refreshCandidate) error {
```

Comment:
this probably needs to take names as well:

if names is empty  then anything in oldHints that is not new hints should be dropped
if names is a list then anything among the names that was in oldHints  but is not in new hints can also be dropped